ToDoReview only for current file, added new argument

marcus-at-localhorst opened the pull request:
```
{
	"keys": ["ctrl+shift+t","ctrl+shift+d"],
	"command": "todo_review",
	"args": {"current_file": true}
}
```

marcus-at-localhorst commented:
PS: This is my first change in a python script and my first pull request ;-)

jonathandelgado commented:
This seems like a good feature to have! Thank you for your contribution and congrats on your first PR!

I haven't tested the PR, but I see some possible issues, I'll do a review right now.

jonathandelgado commented on TodoReview.py:
```diff
@@ -150,6 +150,8 @@ def run(self, edit, **args):
 		window = self.view.window()
 		paths = args.get('paths', None)
 		settings = Settings(self.view, args.get('settings', False))
+		if not paths and not settings.get('current_file', False):
```
I think this needs to be `args.get` rather than `settings.get` just based on line 151 and 155. The settings object will grab from the JSON config while the args object gets specific run-time parameters, like from the keyboard config you referenced.

jonathandelgado replied:
I think we can drop the `if not paths` on this. My main train of though is this argument will only be `True` if it is bound to a key, so the user has already explicitly asked for the current file only. Let me know if you disagree.

marcus-at-localhorst replied:
You are correct, `if not paths` is redundant here. It should be dropped.

jonathandelgado commented:
@marcus-at-localhorst not sure if you missed it, could you fix `settings.get` to be `args.get` as well? I made a comment in that last review with more info.

FichteFoll commented on TodoReview.py:
```diff
@@ -150,6 +150,8 @@ def run(self, edit, **args):
 		window = self.view.window()
 		paths = args.get('paths', None)
 		settings = Settings(self.view, args.get('settings', False))
+		if not settings.get('current_file', False):
```
Logic appears to be reversed here. Change to `if args.get('current_file', False):`.

It also seems that this argument should have the highest priority, so all the following checks should not take place when this is `True`.

FichteFoll commented on TodoReview.py:
```diff
@@ -150,6 +150,8 @@ def run(self, edit, **args):
 		paths = args.get('paths', None)
 		settings = Settings(self.view, args.get('settings', False))
+		if not settings.get('current_file', False):
+			paths = [self.view.file_name()]
```
`view.file_name()` can be `None` if the file is unsaved. I don't think the current engine can handle this case, so just returning and doing nothing would probably be appropriate.

jonathandelgado replied:
Yeah, that is actually a good point. I hadn't considered that scenario. I agree completely. Thanks for the review @FichteFoll

marcus-at-localhorst replied:
Thanks for the hints, that was really helpful. I wonder why my first version even worked :).
I made the check for the `current_file` setting top priority and if the file was not saved I print a note to the console and leave the function.
I left the rest of the logic (`open_files_only` etc) untouched and just moved it into the `else` branch of the `current_file` check.

marcus-at-localhorst commented:
Updated, changed the logic, check if open file exists. See: https://github.com/jonathandelgado/SublimeTodoReview/pull/146/files/08ab2f0ba419df70f7a6f8548565f5292211094d#r96184026

jonathandelgado commented:
LGTM, thanks for the contribution @marcus-at-localhorst! Thanks again for the review, @FichteFoll.
